slurm_ops_manager/nvidia.py

```diff
@@ -0,0 +1,167 @@
+        else:
+            logger.debug("#### Nvidia - setting default repo")
+            # instructions from nvdia website:
+            # https://docs.nvidia.com/datacenter/tesla/tesla-installation-notes/index.html
+
+            if self._operating_system == "ubuntu":
+                distro = "ubuntu2004"
+                repo = f"deb http://developer.download.nvidia.com/compute/cuda/repos/{distro}/x86_64 /" # noqa
+                Path(self._stored.gpu_repo_path).write_text(repo)
+
+                # download pin file
+                url = f"https://developer.download.nvidia.com/compute/cuda/repos/{distro}/x86_64/cuda-{distro}.pin" # noqa
+                path = "/etc/apt/preferences.d/nvidia_gpu-pin-600"
+                cmd = f"curl -s -L -o {path} {url}"
+                subprocess.run(shlex.split(cmd))
+
+                # add key
+                cmd = f"apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/{distro}/x86_64/7fa2af80.pub" # noqa
+                subprocess.run(shlex.split(cmd))
+
+                # it is necessary to update apt's db after every repo change
+                subprocess.call(["apt-get", "update"])
+            elif self._operating_system == "centos":
+                distro = "rhel7"
+                url = f"http://developer.download.nvidia.com/compute/cuda/repos/{distro}/x86_64/cuda-{distro}.repo" # noqa
+                cmd = f"yum-config-manager --add-repo {url}"
+                subprocess.run(shlex.split(cmd))
+
+                subprocess.call(["yum", "clean", "expire-cache"])
+
+        self._stored.gpu_repo_configured = True
+
+    @property
+    def package(self) -> str:
+        """Return nVdia package name."""
+        return self._stored.gpu_package_name
+
+    @package.setter
+    def package(self, pkg: str):
+        """Change the package to install."""
+        if not pkg:
+            # if empty, sets the default
+            if self._operating_system == "ubuntu":
+                self._stored.gpu_package_name = "cuda-drivers"
+            elif self._operating_system == "centos":
+                self._stored.gpu_package_name = "nvidia-driver-latest-dkms cuda cuda-drivers"
+        else:
+            self._stored.gpu_package_name = pkg
+
+    @property
+    def version(self):
+        """Return installed driver version."""
+        if self.installed:
+            pkg = self.package
+
+            if "ubuntu" == self._operating_system:
+                cmd = f"dpkg --status {pkg} | grep \"^Version\""
+                version = subprocess.check_output(cmd, shell=True)
+            elif "centos" == self._operating_system:
+                cmd = f"yum info -C {pkg} | grep \"^Version\""
+                locale = {"LC_ALL": "C", "LANG": "C.UTF-8"}
+                version = subprocess.check_output(cmd, shell=True, env=locale)
+            else:
+                logger.error("## Unsupported OS")
+                return ""
```

Comment:
Yes. The same problems as using `goto`: they make the code exit at unpredictable places. I prefer to avoid both and check the output of the functions.

What would be the advantage of using an exception here?